internal/controller/core/pod_controller.go

```diff
@@ -0,0 +1,263 @@
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type PodReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Create a builder that only watch OpenDataHub global certificate ConfigMap
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Pod{}).
+		Named("core-pod").
+		WithEventFilter(reconcileServingPod()).
+		Complete(r)
+}
+
+// reconcileOpenDataHubGlobalCertConfigMap filters out all ConfigMaps that are not the OpenDataHub global certificate ConfigMap.
+func reconcileServingPod() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			pod := e.Object.(*corev1.Pod)
+			return checkMultiNodePod(pod) && checkPodHasIP(nil, pod)
+		},
+
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			old := e.ObjectOld.(*corev1.Pod)
+			new := e.ObjectNew.(*corev1.Pod)
+			return checkMultiNodePod(new) && checkPodHasIP(old, new)
+		},
+
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+	}
+}
+
+func checkPodHasIP(oldPod *corev1.Pod, newPod *corev1.Pod) bool {
+	if oldPod != nil {
+		return !(oldPod.Status.PodIP == newPod.Status.PodIP)
+	}
+
+	return newPod.Status.PodIP != ""
+}
+
+func checkMultiNodePod(pod *corev1.Pod) bool {
+	for _, contatiner := range pod.Spec.Containers {
+		for _, env := range contatiner.Env {
+			return env.Name == "RAY_USE_TLS" && env.Value != "0"
+		}
+	}
+
+	return false
+}
+
+func (r *PodReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	// Initialize logger format
+	logger := log.FromContext(ctx).WithValues("Pod", req.Name, "namespace", req.Namespace)
+
+	controllerNs := os.Getenv("POD_NAMESPACE")
+
+	pod := &corev1.Pod{}
+	if err := r.Client.Get(ctx, req.NamespacedName, pod); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		logger.Error(err, "Failed to get Pod", "pod", req.Name, "namespace", req.Namespace)
+		return reconcile.Result{}, err
+	}
+	err := reconcileRoleBinding(r.Client, logger, req.Namespace, pod)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = r.reconcileRayTls(ctx, logger, controllerNs, req.Namespace, pod)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func reconcileRoleBinding(k8sClient client.Client, logger logr.Logger, targetNamespace string, pod *corev1.Pod) error {
+	if pod.Spec.ServiceAccountName != "default" {
```

Comment:
so, if for some reason the default SA is used, either passed by the user in the isvc or by accident (if this is possible).
The role binding will be ignored?